### PR TITLE
Crash service support

### PR DESCRIPTION
**Pull request by keianhzo**
Crash handler support.

### CODE REVIEW

File: app/src/common/shared/org/mozilla/vrbrowser/CrashReporterService.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                public void run() {
+                    boolean otherProcessesFound = false;
+                    for (final ActivityManager.RunningAppProcessInfo info : activityManager.getRunningAppProcesses()) {
+                        if (pid != info.pid) {
+                            otherProcessesFound = true;
+                        }
+                    }
+
+                    if (!otherProcessesFound) {
+                        Intent restartIntent = intent.setClass(CrashReporterService.this, VRBrowserActivity.class);
```
Comment (bluemarvin) [inline app/src/common/shared/org/mozilla/vrbrowser/CrashReporterService.java]: I don't understand this part. From the docs intent.setClass() returns it self. So I don't know why you need to reset all the values it already has.

```
intent.setClass(CrashReporterService.this, VRBrowserActivity.class).addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
startActivity(intent);
```

Should be the same?

**Comment by bluemarvin** [comment]
Unfortunately we can't land this until we figure out a work around for the Go crash on first run issue.

File: app/src/common/shared/org/mozilla/vrbrowser/CrashReporterService.java (hunk 1 above)
Comment (keianhzo) [reply app/src/common/shared/org/mozilla/vrbrowser/CrashReporterService.java]: The original intent it targeted to the CrashReporterService so we need to retarget it for the VRBActivity but you are right we don't need to reset the rest of the values. Changed that.

**Comment by bluemarvin** [comment]
@keianhzo I pushed some fixes for Android P to this PR and made the crash reporter setting only grayed out on Oculus. Please take a look.

**Comment by keianhzo** [comment]
@bluemarvin what's the issue with Android P? There is an infinite while loop in the latest committhat keeps on restarting the app whenever it's is closed after the first crash. We should always let the crash service finish after handling the crash intent.

**Comment by bluemarvin** [comment]
I posted a fix for gecko: https://bugzilla.mozilla.org/show_bug.cgi?id=1493227
We should probably wait to land this until the fix is in Gecko or we have an alternative.

**Comment by keianhzo** [comment]
@bluemarvin added a fix for Android P. I still can see issues but only in API level 26.

**Comment by bluemarvin** [comment]
> @bluemarvin added a fix for Android P. I still can see issues but only in API level 26.

What issues are you seeing with 26? My S8 is running O right now.

**Comment by keianhzo** [comment]
@bluemarvin it seems that system applies quite aggressive restrictions when the battery saver is on and the crash reporter service is killed upon second execution.

**Comment by keianhzo** [comment]
@bluemarvin If it's working fine I guess we could land this and file a follow-up for testing in battery saving scenarios as I think this will only affect to Daydream "low end" clip-ins (S8).

**Comment by MortimerGoro** [comment]
We need to wait for https://phabricator.services.mozilla.com/D7282 to land this

**Pull request merged**